Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -272,13 +273,13 @@ extension _BSONEncoder {
         return number
     }
 
-    fileprivate func box<T: Encodable>(_ value: T) throws -> BSONValue? {
+    fileprivate func box<T: Encodable>(_ value: T) throws -> BSONValue {
         // if it's already a `BSONValue`, just return it, unless if it is an
         // array. technically `[Any]` is a `BSONValue`, but we can only use this
         // short-circuiting if all the elements are actually BSONValues.
         if let bsonValue = value as? BSONValue, !(bsonValue is [Any]) {
             return bsonValue
-        } else if let bsonArray = value as? [BSONValue?] {
+        } else if let bsonArray = value as? [BSONValue] {
```

Comment:
Is it actually redundant? Because the first `if` check checks to make sure that the given value is _not_ `[Any]`. That means that even if the value was `[BSONValue]`, it is still also `[Any]`, and would fail the first check.

Then the difference becomes that in the case of `[BSONValue]`, we get back a `[BSONValue]` as opposed to a `BSONValue`. Unsure if perhaps this doesn't make a difference since the return value of the function is a `BSONValue`...

Reply:
I wasn't looking at the condition expressions themselves, but rather the fact that the first branch returns. Therefore, `else if` can simply be written as its own `if` block (keeping the same conditional expression).

Reply:
Ah, I gotcha. Done!